Model/UMEPositionMedal.as

```diff
@@ -0,0 +1,152 @@
+// Ultimate Medals Extended - General Medal representing a position entry
+
+#if DEPENDENCY_ULTIMATEMEDALSEXTENDED
+
+class PositionMedal : UltimateMedalsExtended::IMedal {
+
+    int position;
+    PositionData@ positionData;
+    int offset;
+
+    PositionMedal(int position, PositionData@ positionData, int offset = 0) {
+        this.position = position;
+        @this.positionData = positionData;
+        this.offset = offset;
+    }
+
+    UltimateMedalsExtended::Config GetConfig() override {
+        UltimateMedalsExtended::Config c;
+        c.defaultName = "Top " + this.position;
+
+        if(exportIcon || exportIconColor) {
+            c.icon = positionData.GetColorIcon();
+        }
+
+        if(exportTextColor) {
+            c.nameColor = positionData.textColor;
+        }
+
+        c.sortPriority = 191 - offset; // 191 is the default for position medals, after that a worse position is below (so 191 beats 190)
+
+        // If neither icon nor icon color are exported, use the overlay icon
+        bool useOverlay = !(exportIcon || exportIconColor);
+        c.usePreviousColor = !exportIconColor;
+        c.usePreviousIcon = !exportIcon;
+        c.usePreviousOverlayColor = useOverlay;
+        c.usePreviousOverlayIcon = useOverlay;
+
+        c.shareIcon = false;
+        return c;
+    }
+
+    void UpdateMedal(const string &in uid) override {}
+
+    bool HasMedalTime(const string &in uid) override {
+        LeaderboardEntry@ entry = GetEntry();
+        return (entry !is null && entry.time > 0);
+    }
+
+    uint GetMedalTime() override {
+        LeaderboardEntry@ entry = GetEntry();
+        return uint(entry.time);
+    }
+
+    LeaderboardEntry@ GetEntry() {
+        for(uint i = 0; i < leaderboardArray.Length; i++){
+            if(leaderboardArray[i].position == position && leaderboardArray[i].entryType == EnumLeaderboardEntryType::POSITION){
+                // if we get an entry, update the position data to get the latest values
+                if(leaderboardArray[i].positionData !is null){
+                    positionData = leaderboardArray[i].positionData;
+                }
+                return leaderboardArray[i];
+            }
+        }
+        return null;
+    }
+}
+
+
+array<PositionMedal@> UMEMedals = {};
+
+
+void ClearUME() {
+    for(uint i = 0; i < UMEMedals.Length; i++){
+        UltimateMedalsExtended::RemoveMedal(UMEMedals[i].GetConfig().defaultName);
+    }
+    UMEMedals.Resize(0);
+}
+
+
+void RefreshUME(bool force = false) {
+    if(!exportToUME){
+        ClearUME();
+        return;
+    }
+
+    for(int i = UMEMedals.Length - 1; i >= 0; i--){
+        if(force){
```

Comment:
only use for the force param. it's technically overkill, but I don't want to store the setting's value at creation in the medal, then update them if the value's different from the plugin's value (since you only gather the config once)